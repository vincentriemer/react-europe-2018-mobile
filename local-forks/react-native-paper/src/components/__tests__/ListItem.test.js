/* @flow */

import * as React from 'react';
import renderer from 'react-test-renderer';
import { Text } from 'react-native';
import ListItem from '../ListItem';

it('renders list item with title and description', () => {
  const tree = renderer
    .create(
      <ListItem title="First Item" description="Description for first item" />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders list item with icon', () => {
  const tree = renderer
    .create(<ListItem title="First Item" icon="folder" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders list item with avatar', () => {
  const tree = renderer
    .create(<ListItem title="First Item" avatar={<Text>GG</Text>} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders list item with avatar and icon', () => {
  const tree = renderer
    .create(
      <ListItem
        title="First Item"
        description="Item description"
        icon="folder"
        avatar={<Text>GG</Text>}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
