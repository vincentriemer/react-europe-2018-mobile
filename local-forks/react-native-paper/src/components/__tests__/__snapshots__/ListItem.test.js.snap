// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list item with avatar 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "padding": 8,
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "margin": 8,
          },
          Object {
            "width": 40,
          },
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        GG
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "margin": 8,
          },
          Object {
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Helvetica Neue",
            },
            Array [
              Object {
                "fontSize": 16,
              },
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
              },
            ],
          ]
        }
        theme={
          Object {
            "colors": Object {
              "accent": "#ff4081",
              "background": "#fafafa",
              "disabled": "rgba(0, 0, 0, 0.26)",
              "paper": "#ffffff",
              "placeholder": "rgba(0, 0, 0, 0.38)",
              "primary": "#3f51b5",
              "text": "#000000",
            },
            "dark": false,
            "fonts": Object {
              "light": "HelveticaNeue-Light",
              "medium": "HelveticaNeue-Medium",
              "regular": "Helvetica Neue",
              "thin": "HelveticaNeue-Thin",
            },
            "roundness": 2,
          }
        }
      >
        First Item
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders list item with avatar and icon 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "padding": 8,
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "margin": 8,
          },
          Object {
            "width": 40,
          },
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        GG
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "margin": 8,
          },
          Object {
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Helvetica Neue",
            },
            Array [
              Object {
                "fontSize": 16,
              },
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
              },
            ],
          ]
        }
        theme={
          Object {
            "colors": Object {
              "accent": "#ff4081",
              "background": "#fafafa",
              "disabled": "rgba(0, 0, 0, 0.26)",
              "paper": "#ffffff",
              "placeholder": "rgba(0, 0, 0, 0.38)",
              "primary": "#3f51b5",
              "text": "#000000",
            },
            "dark": false,
            "fonts": Object {
              "light": "HelveticaNeue-Light",
              "medium": "HelveticaNeue-Medium",
              "regular": "Helvetica Neue",
              "thin": "HelveticaNeue-Thin",
            },
            "roundness": 2,
          }
        }
      >
        First Item
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={2}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Helvetica Neue",
            },
            Array [
              Object {
                "fontSize": 14,
              },
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
              },
            ],
          ]
        }
        theme={
          Object {
            "colors": Object {
              "accent": "#ff4081",
              "background": "#fafafa",
              "disabled": "rgba(0, 0, 0, 0.26)",
              "paper": "#ffffff",
              "placeholder": "rgba(0, 0, 0, 0.38)",
              "primary": "#3f51b5",
              "text": "#000000",
            },
            "dark": false,
            "fonts": Object {
              "light": "HelveticaNeue-Light",
              "medium": "HelveticaNeue-Medium",
              "regular": "Helvetica Neue",
              "thin": "HelveticaNeue-Thin",
            },
            "roundness": 2,
          }
        }
      >
        Item description
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "margin": 8,
          },
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders list item with icon 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "padding": 8,
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "margin": 8,
          },
          Object {
            "width": 40,
          },
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "margin": 8,
          },
          Object {
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Helvetica Neue",
            },
            Array [
              Object {
                "fontSize": 16,
              },
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
              },
            ],
          ]
        }
        theme={
          Object {
            "colors": Object {
              "accent": "#ff4081",
              "background": "#fafafa",
              "disabled": "rgba(0, 0, 0, 0.26)",
              "paper": "#ffffff",
              "placeholder": "rgba(0, 0, 0, 0.38)",
              "primary": "#3f51b5",
              "text": "#000000",
            },
            "dark": false,
            "fonts": Object {
              "light": "HelveticaNeue-Light",
              "medium": "HelveticaNeue-Medium",
              "regular": "Helvetica Neue",
              "thin": "HelveticaNeue-Thin",
            },
            "roundness": 2,
          }
        }
      >
        First Item
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders list item with title and description 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "padding": 8,
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "margin": 8,
          },
          Object {
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Helvetica Neue",
            },
            Array [
              Object {
                "fontSize": 16,
              },
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
              },
            ],
          ]
        }
        theme={
          Object {
            "colors": Object {
              "accent": "#ff4081",
              "background": "#fafafa",
              "disabled": "rgba(0, 0, 0, 0.26)",
              "paper": "#ffffff",
              "placeholder": "rgba(0, 0, 0, 0.38)",
              "primary": "#3f51b5",
              "text": "#000000",
            },
            "dark": false,
            "fonts": Object {
              "light": "HelveticaNeue-Light",
              "medium": "HelveticaNeue-Medium",
              "regular": "Helvetica Neue",
              "thin": "HelveticaNeue-Thin",
            },
            "roundness": 2,
          }
        }
      >
        First Item
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={2}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Helvetica Neue",
            },
            Array [
              Object {
                "fontSize": 14,
              },
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
              },
            ],
          ]
        }
        theme={
          Object {
            "colors": Object {
              "accent": "#ff4081",
              "background": "#fafafa",
              "disabled": "rgba(0, 0, 0, 0.26)",
              "paper": "#ffffff",
              "placeholder": "rgba(0, 0, 0, 0.38)",
              "primary": "#3f51b5",
              "text": "#000000",
            },
            "dark": false,
            "fonts": Object {
              "light": "HelveticaNeue-Light",
              "medium": "HelveticaNeue-Medium",
              "regular": "Helvetica Neue",
              "thin": "HelveticaNeue-Thin",
            },
            "roundness": 2,
          }
        }
      >
        Description for first item
      </Text>
    </View>
  </View>
</View>
`;
